### PR TITLE
Add `cross` to GitHub Actions

### LINKED ISSUE #118: Add Pi build to CI using cross
**Is your feature request related to a problem? Please describe.**
Making sure the code builds for pi (arm arch) 

**Describe the solution you'd like**
Install cross as part of CI/CD and cross-compile for pi (aarch64 arch) on one of the supported OS in the matrix

**Describe alternatives you've considered**

**Additional context**
It would be good to find someone offering remote testing on a Pi, but sounds difficult.

### PR DESCRIPTION
**Pull request by sundaram123krishnan**
Fixes #118

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -60,4 +62,28 @@ jobs:
...
\ No newline at end of file
+          verbose: true
+  
+  cross-build-and-test-for-aarch64-unknown-linux-gnu:
+    name: cross-build-and-test-for-aarch64-unknown-linux-gnu
+    runs-on: ubuntu-latest
+    
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
```
Comment (andrewdavidmackenzie) [inline .github/workflows/build_and_test.yml:73]: Use checkoutv4 like in matrix above?

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -60,4 +62,28 @@ jobs:
           file: lcov.info
           flags: unittests
           name: codecov-umbrella
-          verbose: true
\ No newline at end of file
+          verbose: true
+  
+  cross-build-and-test-for-aarch64-unknown-linux-gnu:
```
Comment (andrewdavidmackenzie) [inline .github/workflows/build_and_test.yml:67]: Not a MUST, but what about trying to add this into the above matrix, using " if runner == linux", and then doing the extra step?

That would:
- avoid an extra checkout
- avoid an extra toolchain install (add the target when on linux)

making it all faster, plus

- we could try switching or removing the "if runner == macos" and then get coverage measured for "pi_hw" also...

Sorry for not thinking of that earlier.....

if it got problematic for some reason, we can do it like this though.

**Review by andrewdavidmackenzie — CHANGES_REQUESTED** [review]
try these changes to make it faster / better?

I'll try a quick PR in parallal to see if coverage works on linux also...

**Comment by andrewdavidmackenzie** [comment]
@sundaram123krishnan 
See https://github.com/andrewdavidmackenzie/pigg/pull/238#issuecomment-2235797887

**Review by sundaram123krishnan — COMMENTED** [review]
fixed

File: .github/workflows/build_and_test.yml [hunk 3]
```diff
@@ -43,13 +43,30 @@ jobs:
...
+        run: make cross-test
+
       # lcov on ubuntu-latest (22.04) is still version 1.14 that doesn't have the "--erase-functions" option we want
       - name: clean up coverage Linux
         if: runner.os == 'Linux'
         run: |
           grcov . --binary-path target/debug/ -s . -t lcov --branch --ignore-not-existing --ignore "/*" -o lcov.info
           lcov --remove lcov.info -o lcov.info
-          
+
```
Comment (andrewdavidmackenzie) [inline .github/workflows/build_and_test.yml:69]: Need to rebase? This should do coverage for Linux too

**Review by andrewdavidmackenzie — CHANGES_REQUESTED** [review]

File: .github/workflows/build_and_test.yml (hunk 3 above)
Comment (sundaram123krishnan) [reply .github/workflows/build_and_test.yml:69]: i had rebased ur branch

**Pull request merged**